backend/handlers/user.go

```diff
@@ -91,3 +91,47 @@ func HandleLogin(c *fiber.Ctx) error {
 
 	return c.JSON(fiber.Map{"token": tokenString, "role": user.Role})
 }
+
+func GetUser(c *fiber.Ctx) error {
+	user, ok := c.Locals("userID").(string)
+	if !ok {
+		return c.Status(500).SendString("Invalid userID in header")
+	}
+	if !database.IsValidID(user) {
+		return c.Status(500).SendString("Malformed userID in header")
+	}
+	userID := database.NewID(user)
```

Comment:
This is a bunch of lines that is repeated throughout the backend code. I thought about creating a single `func getLocalUser(c *fiber.Ctx)` that would fetch the current user and parse it. Type system can be leveraged in a way that validation and parsing are done in a single operation that does not lose information, instead of splitting it in two tasks.

This refactor was not part of this user story, and I opted to copy-paste code instead of reinventing the wheel.

Reply:
That makes perfect sense, especially since the refactor wasn't part of the user story. Creating a new function sounds like a good improvement for the future to reduce repetition. For now, the current approach works, and we can always revisit this when we have time to optimise!